evaluator/evaluator/routers/dataset.py

```diff
@@ -1,22 +1,260 @@
+                "value": {
+                    "dataset_name": "commensense_qa",
+                    "skill_type": "multiple-choice",
+                    "metric": "accuracy",
+                    "mapping": {
+                        "id": "id",
+                        "question": "question",
+                        "choices": ["choice.text"],
+                        "answer_index": "1",
+                    },
+                },
+            },
+            "extractive_qa": {
+                "summary": "Extractive sample",
+                "description": "Dataset for Extractive question answering",
+                "value": {
+                    "dataset_name": "quoref",
+                    "skill_type": "extractive-qa",
+                    "metric": "squad",
+                    "mapping": {
+                        "id": "id",
+                        "question": "question",
+                        "context": "context",
+                        "answers": ["answers.text"],
+                    },
+                },
+            },
+        },
+    ),
+):
+    dataset_handler = DatasetHandler()
+    # validate the skill_type if the skill_type
+    if dataset.skill_type != valid_skill_type[dataset.skill_type]:
+        raise HTTPException(
+            404,
+            f"skill_type: {dataset.skill_type} is not supported for our case ",
+        )
+
+    # Load the metric from HuggingFace
+    try:
+        # check if the metric exist on huggingface, when not exist raise exception
+        load(dataset.metric)
+    except FileNotFoundError:
+        raise HTTPException(
+            404, f"Metric with name: {dataset.metric} not found on Huggingface"
+        )
+
+    # check if the dataset_name exist on huggingface
+    try:
+        dataset_handler.download_dataset(dataset.dataset_name)
+
+    except Exception as exception:
+        raise HTTPException(
+            404,
+            f"The Name of the dataset : {dataset.dataset_name} not found! Error {exception}",
+        )
+
+    # check if the dataset_name exist on the database
+    try:
+        if mongo_client.client.evaluator.datasets.find_one(
+            {"dataset_name": dataset.dataset_name}
+        ):
+            myquery = {"dataset_name": dataset.dataset_name}
+            new_value = {
+                "$set": {
+                    "skill_type": dataset.skill_type,
+                    "metric": dataset.metric,
+                    "mapping": {
+                        "id": dataset.mapping.id,
+                        "question": dataset.mapping.question,
+                        "context": dataset.mapping.context,
+                        "answers": dataset.mapping.answers,
+                    },
+                }
+            }
+            mongo_client.client.evaluator.datasets.update_one(myquery, new_value)
+            # download the dataset from hugging_face
+            dataset_handler.get_dataset(dataset.dataset_name)
```

Comment:
Wieso wird oben in der gleichen Funktion download_dataset aufgerufen und hier nochmal get_dataset? 2x ist unnötig